collector/test/CollectorConfigTest.cpp

```diff
@@ -145,4 +151,64 @@ TEST(CollectorConfigTest, TestEnableExternalIpsRuntimeConfig) {
   EXPECT_TRUE(config.EnableExternalIPs());
 }
 
+TEST(CollectorConfigTest, TestYamlConfigToConfigMultiple) {
+  std::vector<std::tuple<std::string, bool, bool>> tests = {
+      {R"(
+                  networkConnectionConfig:
+                    enableExternalIps: true
+               )",
+       true, true},
+      {R"(
+                  networkConnectionConfig:
+                    enableExternalIps: false
+               )",
+       false, true},
+      {R"(
+                  networkConnectionConfig:
+               )",
+       false, true},
+      {R"(
+                  networkConnectionConfig:
+                    unknownFields: asdf
+               )",
+       false, true},
+      {R"(
+                  unknownFields: asdf
+               )",
+       false, false}};
+
+  for (const auto& [yamlStr, expected, valid] : tests) {
+    YAML::Node yamlNode = YAML::Load(yamlStr);
+
+    MockCollectorConfig config;
+
+    bool result = config.MockYamlConfigToConfig(yamlNode);
+    std::optional<sensor::CollectorConfig> runtime_config;
+    runtime_config = config.GetRuntimeConfig();
+
+    EXPECT_EQ(result, valid);
+    EXPECT_EQ(runtime_config.has_value(), valid);
+
+    if (runtime_config.has_value()) {
+      const auto& cfg = runtime_config.value();
+      const auto& network_cfg = cfg.network_connection_config();
+      EXPECT_EQ(network_cfg.enable_external_ips(), expected);
+      EXPECT_EQ(config.EnableExternalIPs(), expected);
+    }
+  }
+}
+
+TEST(CollectorConfigTest, TestYamlConfigToConfigEmpty) {
+  std::string yamlStr = R"()";
+  YAML::Node yamlNode = YAML::Load(yamlStr);
+
+  MockCollectorConfig config;
+
+  EXPECT_DEATH({ config.MockYamlConfigToConfig(yamlNode); }, ".*");
```

Comment:
Why do we need to check the value of runtime_config after this statement? `EXPECT_DEATH` means the program has already stopped, nothing will be done with the value.

Reply:
Removed checks after this.